sc-memory/sc-memory/sc_iterator.hpp

```diff
@@ -78,27 +83,39 @@ class TIterator3 : public TIteratorBase<sc_iterator3>
     if (m_iterator)
     {
       sc_iterator3_free(m_iterator);
-      m_iterator = 0;
+      m_iterator = nullptr;
     }
   }
 
-  _SC_EXTERN bool Next() const
+  _SC_EXTERN bool Next() const override
   {
-    SC_ASSERT(IsValid(), ());
+    SC_ASSERT(IsValid(), ("Not valid iterator object"));
     return sc_iterator3_next(m_iterator) == SC_TRUE;
   }
 
-  _SC_EXTERN ScAddr Get(sc_uint8 idx) const
+  _SC_EXTERN ScAddr Get(size_t index) const override
   {
-    SC_ASSERT(idx < 3, ());
-    SC_ASSERT(IsValid(), ());
-    return ScAddr(sc_iterator3_value(m_iterator, idx));
+    SC_ASSERT(IsValid(), ("Not valid iterator object"));
```

Comment:
Please change to exception

Reply:
SC_ASSERT is exception, I propose to leave it here, because it check if used context from we use concrete method is valid. In other cases with check and call other exceptions (f.e. after not successfull params checking)

Reply:
sc_assert is not exception